Sources/xcodeproj/Objects/Project/PBXProj.swift

```diff
@@ -200,6 +205,9 @@ extension PBXProj: Equatable {
 extension PBXProj: Writable {
     public func write(path: AbsolutePath, override: Bool) throws {
         let encoder = PBXProjEncoder()
+        encoder.fileListSortOrder = fileListSortOrder
```

Comment:
Instead of using an instance variable that is set to the encoder like this, I'd pass is as an argument to the `write` and to the `encode` methods. To avoid having to pass three variables, you can define a new type that contains the sorting order for each of those three elements:

```swift
struct PBXProjEncoderOrder {
  let files: FileListSortOrder
  let navigator: ProjectNavigatorSortOrder
  let buildPhase: BuildPhaseFileSortOrder
}
```
Then you can update the write method signature to be like:

```swift
// Pass a default order value
public func write(path: AbsolutePath, override: Bool, order: PBXProjEncoderOrder = PBXProjEncoderOrder()) throws
```

Reply:
Nice. Will also build in a default for anyone which implements the defaults previously used.